sympy/sets/tests/test_fancysets.py

```diff
@@ -441,6 +441,24 @@ def test_imageset_intersect_real():
     assert s.intersect(S.Reals) == imageset(Lambda(n, 2*n*pi - pi/4), S.Integers)
 
 
+def test_imageset_intersect_interval():
+    from sympy.abc import n
+    f1 = ImageSet(Lambda(n, n*pi), S.Integers)
```

Comment:
Please also add tests for imagesets which are not "linear", for example `ImageSet(Lambda(n, log(n)), S.Integers)` and one which are not monotonic like `ImageSet(Lambda(n, n**2), S.Integers)`.